packages/core/strapi/lib/services/entity-service/types/result.d.ts

```diff
@@ -2,24 +2,63 @@ import type { Attribute, Common, EntityService, Utils } from '@strapi/strapi';
 
 type Pagination = { page: number; pageSize: number; pageCount: number; total: number };
 
-export type Entity<TSchemaUID extends Common.UID.Schema> = Utils.Object.DeepPartial<
-  GetValues<TSchemaUID>
+export type Entity<
+  TSchemaUID extends Common.UID.Schema,
+  TParams extends EntityService.Params.Pick<TSchemaUID, 'fields' | 'populate'>
```

Comment:
I'm not sure the best way is the put the params here directly; Entity shouldn't be aware of params at all IMO

I see some alternatives:

1. Change params to the actual list of attributes, so it would be something similar to `Entity<'api::some.schema', AttributesFromParams<'api::some.schema', 'populate' | 'fields'>>`
2. Keep Entity untouched as the full entity, but add something like `EntityQueryResult<TContentTypeUID, TParams>` which would make sense to pass the params.

I think I like option 2 best, because it is clear about what it is and not trying to fit multiple ideas into Entity. It would still be possible for someone to convert it to an Entity by adding missing fields, and vice versa by doing a pick or a typeguard to convert an Entity into an EntityQueryResult.